Fix bashism in config.sh script. This fixes Debian bug #530119. The
implementation of this patch by Daniel Moerner was based on the tips at 
https://wiki.ubuntu.com/DashAsBinSh.

--- smlnj.orig/base/smlnj-lib/Doc/config.sh	2007-05-10 14:39:48.000000000 -0700
+++ smlnj/base/smlnj-lib/Doc/config.sh	2009-09-03 23:39:05.000000000 -0700
@@ -6,7 +6,7 @@
 
 find ML-Doc -name "*.mldoc" -print | mk-mldoc-makefile
 
-function mkDirTree {
+mkDirTree () {
   base=$1
   if test ! -d $base ; then
     mkdir $base || (echo "unable to create $base"; exit 1)
